src/vmm/src/devices/virtio/vhost_user.rs

```diff
@@ -257,3 +464,517 @@ impl VhostUserHandle {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::undocumented_unsafe_blocks)]
+    #![allow(clippy::cast_possible_truncation)]
+
+    use utils::tempfile::TempFile;
+
+    use super::*;
+    use crate::vstate::memory::{FileOffset, GuestAddress, GuestMemoryExtension};
+
+    #[test]
+    fn test_new() {
+        struct MockMaster {
+            sock: UnixStream,
+            max_queue_num: u64,
+            is_owner: std::cell::UnsafeCell<bool>,
+        }
+
+        impl VhostUserHandleBackend for MockMaster {
+            fn from_stream(sock: UnixStream, max_queue_num: u64) -> Self {
+                Self {
+                    sock,
+                    max_queue_num,
+                    is_owner: std::cell::UnsafeCell::new(false),
+                }
+            }
+
+            fn set_owner(&self) -> Result<(), vhost::Error> {
+                unsafe { *self.is_owner.get() = true };
+                Ok(())
+            }
+        }
+
+        let max_queue_num = 69;
+
+        let tmp_dir = utils::tempdir::TempDir::new().unwrap();
+        let tmp_dir_path_str = tmp_dir.as_path().to_str().unwrap();
+        let tmp_socket_path = format!("{tmp_dir_path_str}/tmp_socket");
+
+        unsafe {
+            let socketfd = libc::socket(libc::AF_UNIX, libc::SOCK_STREAM, 0);
+            if socketfd < 0 {
+                panic!("Cannot create socket");
+            }
+            let mut socket_addr = libc::sockaddr_un {
+                sun_family: libc::AF_UNIX as u16,
+                sun_path: [0; 108],
+            };
+
+            std::ptr::copy::<i8>(
+                tmp_socket_path.as_ptr().cast(),
+                socket_addr.sun_path.as_mut_ptr(),
+                tmp_socket_path.as_bytes().len(),
+            );
+
+            let bind = libc::bind(
+                socketfd,
+                (&socket_addr as *const libc::sockaddr_un).cast(),
+                std::mem::size_of::<libc::sockaddr_un>() as u32,
+            );
+            if bind < 0 {
+                panic!("Cannot bind socket");
+            }
+
+            let listen = libc::listen(socketfd, 1);
+            if listen < 0 {
+                panic!("Cannot listen on socket");
+            }
+        }
+
+        let vuh = VhostUserHandleImpl::<MockMaster>::new(&tmp_socket_path, max_queue_num).unwrap();
+        assert_eq!(
+            vuh.vu
+                .sock
+                .peer_addr()
+                .unwrap()
+                .as_pathname()
+                .unwrap()
+                .to_str()
+                .unwrap(),
+            &tmp_socket_path,
+        );
+        assert_eq!(vuh.vu.max_queue_num, max_queue_num);
```

Comment:
let's add a text description of what we are verifying in the tests where it's absent.

Reply:
added